src/main/resources/avro/common.avdl

```diff
@@ -5,44 +5,45 @@ not have any methods; it is merely a library of types.
 
 ##Sequences, Segments, and References
 
-A sequence is a component in a sequence graph, optionally joined at its ends to
-other sequences. Sequences are not directly retrievable through the API, but
-ranges of their bases can be downloaded.
-
-A `Segment` is a range on a sequence, possibly including the joins at the
-sequence's ends. It does not include base data. When you ask about the novel
-sequences that come in a `VariantSet`, you will get back `Segment`s covering
-them. A `Path` through a sequence graph is a list of `Segment`s.
-
-A `Reference` (defined in references.avdl) is an object that marks a sequence as
-part of a `ReferenceSet`'s sequence graph. It contains a `Segment` covering the
-sequence it belongs to.
+A `Sequence` is a component in a sequence graph, optionally joined to other
+`Sequence`s via `Join`s. It has to have length at least 1, but does not contain
+base data itself - the base data is obtained with the `getSequenceBases()`
+method call.
+
+A `Segment` is a range on a `Sequence`. When you ask about the novel
+`Sequence`s that come in a `VariantSet`, you will get back `Segment`s covering
+them. A `Path` through a sequence graph is a list of `Segment`s, with the
+requirement that adjacent segments are joined by `Join`s.
+
+A `Reference` (defined in references.avdl) is part of a `ReferenceSet` which
+represents the reference sequence graph. It contains a `Segment` identifying
+the underlying `Sequence`. The reason we do this rather than directly containing
+the `Sequence` is to allow generation of local `Reference`s and `ReferenceSet`s
+from part of a larger one, e.g. the MHC region of chromosome 6.
 
 ##Sequence Graph Concepts
 
 Many of the types defined here are related to defining sequence graphs. Sequence
 graph concepts can be illustrated on a piece of double-stranded DNA with two
 nicks in it, one on the top strand and one on the bottom strand. Here is an
-example where the top strand is GGTGGNG.
-
-```
-            | <- top strand nick at position (2,+)
-5' -------  -------------------  3'
-     G   G   T   G   G   N   G
-     C   C   A   C   C   N   C
-3' ----------------  ----------  5'
-                  | <- bottom strand nick at position (3,-)
-
-    +0- +1- +2- +3- +4- +5- +6- <- coordinates
-```
-
-A sequence is a piece of double-stranded DNA composed of a series of DNA
-basepairs. In the default forward orientation a sequence is specified by the DNA
-letters of its top strand, e.g. in this case GGTGGNG, where N indicates an
+example where the top strand is GGTGGAG.
+
```

Comment:
Example below still uses GGTGGNG. Do you want to change N->A throughout the rest of the example?